kube-derive/src/lib.rs

```diff
@@ -150,6 +150,18 @@ mod custom_resource;
 /// }
 /// ```
 ///
+/// ## Customizing Schemas
+/// Should you need to customize the schemas, you can use:
+/// - [Serde/Schemars Attributes](https://graham.cool/schemars/examples/3-schemars_attrs/) (no need to duplicate serde renames)
+/// - [`#[schemars(schema_with = "func")]`](https://graham.cool/schemars/examples/7-custom_serialization/) (e.g. like in the [`crd_derive` example](https://github.com/clux/kube-rs/blob/master/examples/crd_derive.rs))
+///
+/// In general, you will need to override parts of the schemas (for fields in question) when you are:
+/// - **using enums**: enums do not currently generate [structural schemas](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#specifying-a-structural-schema), so kubernetes won't support them by default
```

Comment:
nit: simple enums should work (string enums)

Reply:
i'll s/enums/complex enums/